provider/maas/environ.go

```diff
@@ -2134,17 +2141,24 @@ func (env *maasEnviron) allocateContainerAddresses1(hostInstanceID instance.Id,
 		deviceNICIDs[i] = maasNICID
 		subnetID := string(nic.ProviderSubnetId)
 
+		if !knownSubnet {
+			continue
+		}
+
 		linkedInterface, err := env.linkDeviceInterfaceToSubnet(deviceID, maasNICID, subnetID, modeStatic)
 		if err != nil {
-			return nil, errors.Annotate(err, "cannot link device interface to subnet")
+			logger.Warningf("linking NIC %v to subnet %v failed: %v", nic.InterfaceName, nic.CIDR, err)
```

Comment:
It's likely and issue worth investigating, but not necessarily an error that should cause the whole allocation stage to fail. I've seen people using MAAS setups with "public" subnets, which they need configured on the nodes, but are not managed by MAAS and thus it can't provide DHCP or static IPs when a NIC is linked to them (e.g. users are expected to manually obtain and configure statically IPs from such subnets - like a floating IP pool in OpenStack).